### DISCUSSION TITLE
Test for dimensionless object

### DISCUSSION THREAD

**Discussion by nopria** [sympy/sympy#25871] [Q&A]
How can I determine if a sympy object is dimensionless, i.e. is a pure number and not a physical quantity? This is related to issue #14374, the problem arises when the input is zero, or more exactly `sympy.core.numbers.Zero`, which might mean both a zero pure number and 0 cm, 0 kg, 0 newton, etc.. I'm looking for workarounds until the issue is solved.

**Comment by oscarbenjamin** [comment]
Ideally `obj.kind` would give you this.

**Reply by nopria** [reply]
How `sympy.core.numbers.Zero.kind` can dintinguish if zero comes from `0*45` or from `0*(45*meters)` ?

**Reply by oscarbenjamin** [reply]
The check should be used before evaluation.

**Reply by nopria** [reply]
The problem is that it can't. In real coding you have to deal with processes which involve hundreds of multiplications at the end of which you have to check if the value is dimensionless or not in order to proceed with more hundreds of multiplications and so on.

**Reply by oscarbenjamin** [reply]
I mean that the internal code of Mul should check .kind
